File: services/backend/src/main/java/com/gume/sonar/domain/Prompt.java
package com.gume.sonar.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Prompt {
    private UUID id;
    private String title;
    private String content;
    private User user;
    private LocalDateTime lastUpdate;
}

File: services/frontend/src/domain/models/Prompt.ts
import { type Client } from './Client';

export interface Prompt {
  id: string;
  title: string;
  content: string;
  client: Client;
  lastUpdate: string;
}

export interface PromptInput {
  title: string;
  content: string;
  client: Pick<Client, 'id'>;
}
